### PR TITLE
add convenience method - first

### PR DESCRIPTION
**Pull request by drujensen**
This adds a new query method `first` that adds a `LIMIT 1` to the query and returns a single object instead of an array.

This also allows you to pass a SQL clause just like the `all` method to perform the query.

For example:
```
first_post = Post.first("WHERE posts.active = true 
                         ORDER BY posts.updated_at DESC")
```

### CODE REVIEW

File: src/granite_orm/querying.cr [hunk 1]
```diff
@@ -45,6 +45,16 @@ module Granite::ORM::Querying
...
   end
 
+  # First adds a `LIMIT 1` clause to the query and returns the first result
+  def first(clause = "", params = [] of DB::Any)
+    results = all([clause.strip, "LIMIT 1"].join(" "), params)
+    if results.size > 0
+      results.first
+    else
+      nil
+    end
```
Comment (c910335) [inline src/granite_orm/querying.cr]: How about `results[0]?`?

Comment (migeorge) [reply src/granite_orm/querying.cr]: Is this a stylistic choice or is there a performance difference? Just curious

Comment (c910335) [reply src/granite_orm/querying.cr]: We don't need to check size here because `#[]?` did it.

Comment (c910335) [reply src/granite_orm/querying.cr]: Oh I just found a better choice which is `#.first?`.

File: src/granite_orm/querying.cr [hunk 2]
```diff
@@ -45,6 +45,16 @@ module Granite::ORM::Querying
     return rows
   end
 
+  # First adds a `LIMIT 1` clause to the query and returns the first result
+  def first(clause = "", params = [] of DB::Any)
+    results = all([clause.strip, "LIMIT 1"].join(" "), params)
+    if results.size > 0
+      results.first
+    else
+      nil
```
Comment (eliasjpr) [inline src/granite_orm/querying.cr]: Do we need explicitly to return `nil`?

File: src/granite_orm/querying.cr (hunk 1 above)
Comment (drujensen) [reply src/granite_orm/querying.cr]: good call, I will use that

File: src/granite_orm/querying.cr (hunk 2 above)
Comment (drujensen) [reply src/granite_orm/querying.cr]: @c910335 recommended `#first?` which really cleans this up.

File: src/granite_orm/querying.cr [hunk 3]
```diff
@@ -45,6 +45,11 @@ module Granite::ORM::Querying
     return rows
   end
 
+  # First adds a `LIMIT 1` clause to the query and returns the first result
+  def first(clause = "", params = [] of DB::Any)
+    results = all([clause.strip, "LIMIT 1"].join(" "), params).first?
```
Comment (c910335) [inline src/granite_orm/querying.cr]: Now `results` is unnecessary.

Comment (drujensen) [reply src/granite_orm/querying.cr]: right.  thx.

**Pull request merged**